lib/mdx.ml

```diff
@@ -78,7 +79,18 @@ type syntax = Syntax.t =
   | Cram
 
 let parse_lexbuf syntax l = parse (Lexer.token syntax l)
-let parse_file syntax f = parse (Lexer.token syntax (snd (Misc.init f)))
+
+let parse_file ?syntax f =
```

Comment:
This argument doesn't need to be optional.
It is never called without. Also, it adds responsibilities to the function, adds dead code and requires a fallback case (`| None -> Normal`). I think this should be handled at a higher level.

Reply:
This is done for https://github.com/realworldocaml/mdx/pull/217/files#r374035482

Reply:
I think it's the responsibility of that code to call infer and to define the fallback case.

Reply:
Yes we can do that instead